tests/suites/test_suite_ssl.function

```diff
@@ -928,31 +928,86 @@ int mbedtls_move_handshake_to_state( mbedtls_ssl_context *ssl,
 #endif /* MBEDTLS_X509_CRT_PARSE_C */
 
 /*
- * Write application data. Then increase write and fragments counter
+ * Write application data. Increase write counter and fragments counter if
+ * necessary.
  */
 int mbedtls_ssl_write_fragment( mbedtls_ssl_context *ssl, unsigned char *buf,
-                                int ln, int *writen, int *fragments )
+                                int buf_len, int *written,
+                                int *fragments, const int expected_frames )
 {
-    int ret = mbedtls_ssl_write( ssl, buf + *writen, ln - *writen );
-    if( ret >= 0 )
+    int ret = mbedtls_ssl_write( ssl, buf + *written, buf_len - *written );
+    if( ret > 0 )
     {
         (*fragments)++;
-        *writen += ret;
+        *written += ret;
     }
-    return ret;
+
+    if( expected_frames == 0 )
+    {
+        /* Used for DTLS and the message size larger than MFL. This error must
+         * be returned for some of tests */
+        TEST_ASSERT( ret == MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+        return ret;
+    }
+    else if( expected_frames == 1 )
+    {
```

Comment:
`expected_frames` could be unsigned, and then a rough sketch:
```suggestion
else
{
    int size_ok = ( expected_frames == 1 ? ( ret == buf_len ) : ( ret >= 0 && ret <= buf_len )  );
    int transport_not_ready = ( ret == MBEDTLS_ERR_SSL_WANT_READ || ret == MBEDTLS_ERR_SSL_WANT_WRITE );
    TEST_ASSERT( size_ok || transport_not_ready );
}
```

Reply:
For this current code `expected_frames` could be unsigned or even better size_t, but if in the future I would like to use this functions for data transfer only (without checking the number of fragments) I could easy add `expected_frames = -1` as a parameter to exclude checking of fragments. Except that we have many places in the code when we use int and could be unsigned int. It is just simpler and good enough.
I would not say that this option is more elegant. There is need two more variables and when you find some error in the report from this line it will be harder to determine what actually could failed. Tests are not ordinary programs and should be more transparent and understandable than well-optimized. The only one duplication here is double check of `MBEDTLS_ERR_SSL_WANT_READ` and `MBEDTLS_ERR_SSL_WANT_WRITE`